### DISCUSSION TITLE
Gunicorn worker is restarting while sending the large response using fastAPI

### DISCUSSION THREAD

**Discussion by AniketModi** [fastapi/fastapi#5682] [Questions] [answered]
We have a python app. we have implemented the fastAPI in app. App will get request and will return response. While returning large response, gunicorn worker is restarting and not able to send large response. This case is happening when we get multiple requests in parallel.

**Comment by yinziyan1206** [comment, marked as answer]
```python
from fastapi import FastAPI


app = FastAPI()


@app.get("/")
async def get_response():
    return {k: k for k in range(100000)}
```
![image](https://user-images.githubusercontent.com/37829370/203757125-8cc70746-636b-4d14-a83a-b2aa3c885a98.png)
how large the response content is. I try it with 4 workers and seem to be ok

**Reply by AniketModi** [reply]
I am trying with 4 workers already. It is 13 MB as of now and it is expected to grow by days

**Reply by AniketModi** [reply]
What can be the reason that worker is not able to send the response via fastAPI

**Reply by AniketModi** [reply]
And this is happening in POST request. Just fyi

**Reply by yinziyan1206** [reply]
if worker is restarting after starting up, it should be some system error (such as outofmemory).

**Reply by nymous** [reply]
I had a similar issue when using FastAPI with Uvicorn Gunicorn workers, where I created an endpoint using the `async` keyword but containing blocking (non-async) code.
In that case the worker is blocked by the endpoint, and cannot respond to the healthcheck pings from Gunicorn; Gunicorn thinks the worker died and restarts it.
Switching the endpoint to *not* be `async` fixes the problem, as its code is then run in a threadpool, leaving the worker available to respond to pings.

Not saying this is where your issue comes, just another thing to check.

**Reply by AniketModi** [reply]
okay

**Reply by AniketModi** [reply]
I have also tried with async but still I am getting an issue

**Reply by AniketModi** [reply]
I would like to explain in a little bit details::
We have one app, which runs the query and returned the result as a response. We have a 4 workers there. When I hit 3 parallel requests, all 3 are being processed but 2 or 1  of them are  not returning the response and they are restarting

**Reply by AniketModi** [reply]
The app gets the result accessing the database.

**Reply by AniketModi** [reply]
We are using duckDB and database have parquet files
